hack/go/generate_versions.go

```diff
@@ -199,20 +204,50 @@ func getVersion(repo, ref string) (string, error) {
 	return raw, nil
 }
 
-func fetchVersion(path string) (string, error) {
-	resp, err := retryablehttp.Get(path)
-	if err != nil {
-		return "", err
+// fetchVersion returns the first successful response from the provided paths (async).
+func fetchVersion(paths []string) (string, error) {
+	var (
+		version string
+		err     error
+	)
+	versionChan := make(chan string, 1)
+	errChan := make(chan error, 1)
+	ctx, cancel := context.WithCancel(context.Background())
+
+	for _, path := range paths {
+		go func(path string, cancel context.CancelFunc) {
```

Comment:
No need for concurrency here I think, no problem if it takes an extra second.
I'd run it sequentially to keep the code simple/more readable.
/lgtm
so we can move forward, but I'll
/hold
please don't hesitate to unhold.

Reply:
Removed.